mapiproxy/servers/default/emsmdb/oxosfld.c

```diff
@@ -0,0 +1,224 @@
+/*
+   OpenChange Server implementation
+
+   EMSMDBP: EMSMDB Provider implementation
+
+   Copyright (C) Enrique J. Hernández 2015
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+   \file oxosfld.c
+
+   \brief Server-side procedures about special folders
+ */
+
+#include "dcesrv_exchange_emsmdb.h"
+
+#include <gen_ndr/ndr_property.h>
+
+static bool fid_in_entry_id(struct Binary_r *checked_entry_id,
+                            uint64_t parent_checked_entry_id_fid,
```

Comment:
A very weird name, isn't it? This is a "normal" folder id, I'd rename this to just `parent_fid`